core/server/worker/src/test/java/alluxio/worker/grpc/BlockReadHandlerTest.java

```diff
@@ -30,16 +31,28 @@
 import io.netty.util.ResourceLeakDetector;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DefaultBlockWorker.class})
 public final class BlockReadHandlerTest extends ReadHandlerTest {
-  private BlockWorker mBlockWorker;
+  private DefaultBlockWorker mBlockWorker;
   private BlockReader mBlockReader;
 
   @Before
   public void before() throws Exception {
     ResourceLeakDetector.setLevel(ResourceLeakDetector.Level.ADVANCED);
-    mBlockWorker = mock(BlockWorker.class);
+
+    // Use PowerMockito to mock final class and able to test the method contents
+    mBlockWorker = PowerMockito.mock(DefaultBlockWorker.class);
```

Comment:
if  needed, we can make `DefaultBlockWorker` non final to avoid `PowerMockito`

Reply:
Add a TODO for making DefaultBlockWorker non-final and avoid PowerMockito.
The test changes will be given in another PR to unblock the current implementation

Reply:
I took a deep look, I can make `DefaultBlockWorker` not final and use `extends DefaultBlockWorker`, but then in order to create the `MockDefaultBlockWorker`, i need to pass in (mock) classes including `BlockMasterClient.class, BlockMasterClientPool,BlockStore, FileSystemMasterClient,Sessions`. Those classes are all final. 

So in order to avoid using powerMockito, all those classes need to change to non-final. Many other mock classes may be needed. And there are maybe more to change in the next layer.

The whole `BlockWorkerTest` relies on PowerMockito, 
```
@RunWith(PowerMockRunner.class)
@PrepareForTest({BlockMasterClient.class, BlockMasterClientPool.class, FileSystemMasterClient.class,
    BlockHeartbeatReporter.class, BlockMetricsReporter.class, BlockMeta.class,
    BlockStoreLocation.class, StorageDir.class, ServerConfiguration.class, UnderFileSystem.class,
    BlockWorker.class, Sessions.class})
```
I couldn't find an easy way now to reconstruct the test to not using PowerMockito.

Would be better to reconsider how to test the whole `BlockWorker`. `BlockReadHandlerTest` can then extend the logic and change.